### ISSUE TITLE
Add vanilla buttons to forum-toolbar settings

### ISSUE THREAD

**Issue by mybearworld** [ScratchAddons/ScratchAddons#6059]
### Why this would be helpful

A lot of people probably don't use some buttons of the forum toolbar. They may want to get rid of them.

### My suggestion

Add options to `forums-toolbar` to remove the vanilla buttons as well.

### Possible alternatives

We could create a separate addon instead, but this would make more sense.

**Comment by WorldLanguages** [comment]
Sounds like over-customization to me

**Comment by mybearworld** [comment]
> Sounds like over-customization to me

I wouldn't say so. I don't think I'm alone in wanting to remove the Clean button, for example.

**Comment by WorldLanguages** [comment]
Find someone who really wants to remove the Clean button but doesn't know about AdBlock's "hide something on this page" feature or any other alternative way to hide it 🤔

**Comment by mybearworld** [comment]
> Find someone who really wants to remove the Clean button but doesn't know about AdBlock's "hide something on this page" feature or any other alternative way to hide it 🤔

With the same logic, `hide-delete-button`, `hide-project-stats` , `hide-signatures` and `hide-stage` could be removed as well.

**Comment by WorldLanguages** [comment]
I wouldn't say those are directly comparable - all but the last you mentioned hide elements that may appear multiple times per page. And I don't need to elaborate about `hide-stage`, right? You can't achieve the same UX with AdBlock, and hiding the stage with `display:none` causes crashes.

**Comment by WorldLanguages** [comment]
That addon idea is completely obvious and it adds a toggle button which requires JavaScript ¯\_(ツ)_/¯

**Comment by BroJac5246** [comment]
> Sounds like over-customization to me

While I do agree it's a good bit of customization, users (including myself) would definitely use this. I never use a lot of the buttons, particularly since keyboard shortcuts exist, and as a result, it is annoying when I actually do have to find a button to use because I have trouble locating it.

**Comment by WorldLanguages** [comment]
> While I do agree it's a good bit of customization, users (including myself) would definitely use this. I never use a lot of the buttons, particularly since keyboard shortcuts exist, and as a result, it is annoying when I actually do have to find a button to use because I have trouble locating it.

How many of the buttons would you hide? Can't you hide them already through other methods?

**Comment by Samq64** [comment]
I would probably hide the numeric list, user agent, clean and preview buutons (when I have the live preview enabled).

I don't think there's any reason to have the option to hide the first few buttons (bold, italic, underline and strikethrough)

**Comment by mybearworld** [comment]
Would it be better if there was options for "Formatting", "URL/Images", "Font Size", "Lists", "Other Tags", "Browser/OS", "Scratchblocks", "Clean/Preview" (e.g. the sections) instead of all the buttons?

**Comment by WorldLanguages** [comment]
That's probably better but "clean" probably deserves its own setting.

**Comment by mybearworld** [comment]
True, for getting rid of the preview button, `forum-live-preview` already exists anyways.

**Comment by Secret-chest** [comment]
> I don't think there's any reason to have the option to hide the first few buttons (bold, italic, underline and strike through) through.

I believe otherwise, those are easy to make with keyboard shortcuts.

**Comment by BroJac5246** [comment]
> Would it be better if there was options for "Formatting", "URL/Images", "Font Size", "Lists", "Other Tags", "Browser/OS", "Scratchblocks", "Clean/Preview" (e.g. the sections) instead of all the buttons?

"URL" and "Images" should be separate settings because the former is easily and habitually done through a keyboard shortcut, so people will want to hide the "URL" button but not the "Images" button.

**Comment by WorldLanguages** [comment]
I forgot `Forums image uploader` adds its own button, right?

**Comment by DNin01** [comment]
Is there a good enough reason to hide these useful features? How many people care about hiding them?

If someone hid buttons they never used before, they also wouldn't have the chance to explore them.

Also, what if a user forgets they hid a button or did so accidentally and then someone else tells them to use one of the hidden buttons?

> With the same logic, `hide-delete-button`, `hide-project-stats` , `hide-signatures` and `hide-stage` could be removed as well.

Hiding signatures can save a lot more time than hiding forum toolbar buttons. Hiding the delete button prevents many accidents while still allowing you to delete through a right-click.

---

[Related comment](https://github.com/ScratchAddons/ScratchAddons/discussions/4619#discussioncomment-2845820)

**Comment by BroJac5246** [comment]
> Is there a good enough reason to hide these useful features? How many people care about hiding them?
> 
> If someone hid buttons they never used before, they also wouldn't have the chance to explore them.
> 
> Also, what if a user forgets they hid a button or did so accidentally and then someone else tells them to use one of the hidden buttons?
> 
> > With the same logic, `hide-delete-button`, `hide-project-stats` , `hide-signatures` and `hide-stage` could be removed as well.
> 
> Hiding signatures can save a lot more time than hiding forum toolbar buttons. Hiding the delete button prevents many accidents while still allowing you to delete through a right-click.
> 
> ---
> 
> [Related comment](https://github.com/ScratchAddons/ScratchAddons/discussions/4619#discussioncomment-2845820)
> 

I disagree. All options should be disabled by default, so users will be explicitly opting-out of them. And yes, this does save time because you don't need to search through options you don't need. Beyond that, it also simply declutters the toolbar.

**Comment by WorldLanguages** [comment]
> [Related comment](https://github.com/ScratchAddons/ScratchAddons/discussions/4619#discussioncomment-2845820)

Nice find. One year ago, I commented:

> This level of customization (hiding specific, important buttons) can be better customized by users themselves by writing their own userstyles or even using adblocker features that let you hide elements.

I wouldn't consider these forum buttons "important", so if this gets added, I suggest that the number of settings we add stays reasonable, instead of adding one setting for each specific button.
